src/query/storages/parquet/src/parquet2/parquet_reader/data.rs

```diff
@@ -43,37 +44,10 @@ pub trait SeekRead: std::io::Read + std::io::Seek {}
 
 impl<T> SeekRead for T where T: std::io::Read + std::io::Seek {}
 
-pub struct DataReader {
-    bytes: usize,
-    inner: Box<dyn SeekRead + Sync + Send>,
-}
-
-impl DataReader {
```

Comment:
DataReader for block_io is used to make the real reading of data lazy, with pruning, some pages do not to be read. 
I remember it works well when comparing with other db. 
maybe we should keep  this wrapper.

and maybe for blocking io, we can choose to use lazy io and merge_io dependently. io page can be merged, too complicate?

cc @RinChanNOWWW @sundy-li

Reply:
@RinChanNOWWW correct my if I am wrong:

the parquet_rs impl seems abandoned the blocking_io.
merge io can be done in InMemoryRowGroup::fetch. 

so it seem not hard to reuse the merge_io code.

Reply:
> the parquet_rs impl seems abandoned the blocking_io.

no, `async fn get_ranges(&self, ranges: &[Range<u64>]) ` will use blocking io path.